### ISSUE TITLE
etcds stuck in initializing state when 2 hosts are down and two new hosts are added 

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#5480]
**Rancher Version:**: 1.1.1

**OS and where are the hosts located? (cloud, bare metal, etc):**

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**: Kubernetes

**Steps to Reproduce:**
1. Create a 3 host Kubernetes set up with rancher-catalog
2. Turn down two hosts
3. Add 2 hosts
4.  Delete the two hosts powered down
5. Perform Disaster recovery

**Results**
1. etcds get created on both the hosts, but they are stuck in initializing state.

etcd logs on the newly added host
=========================

```
7/19/2016 9:52:00 AMGet http://Kubernetes_etcd_1:2379/health: dial tcp: lookup Kubernetes_etcd_1 on 169.254.169.250:53: no such host
7/19/2016 9:52:01 AMGet http://Kubernetes_etcd_1:2379/health: dial tcp: lookup Kubernetes_etcd_1 on 169.254.169.250:53: no such host
7/19/2016 9:52:03 AMGet http://Kubernetes_etcd_1:2379/health: dial tcp: lookup Kubernetes_etcd_1 on 169.254.169.250:53: no such host
7/19/2016 9:52:04 AMGet http://Kubernetes_etcd_1:2379/health: dial tcp: lookup Kubernetes_etcd_1 on 169.254.169.250:53: no such host
7/19/2016 9:52:06 AMGet http://Kubernetes_etcd_1:2379/health: dial tcp: lookup Kubernetes_etcd_1 on 169.254.169.250:53: no such host
7/19/2016 9:52:08 AMGet http://Kubernetes_etcd_1:2379/health: dial tcp: lookup Kubernetes_etcd_1 on 169.254.169.250:53: no such host
7/19/2016 9:52:10 AMGet http://Kubernetes_etcd_1:2379/health: dial tcp: lookup Kubernetes_etcd_1 on 169.254.169.250:53: no such host
7/19/2016 9:52:13 AMGet http://Kubernetes_etcd_1:2379/health: dial tcp: lookup Kubernetes_etcd_1 on 169.254.169.250:53: no such host
7/19/2016 9:52:15 AMGet http://Kubernetes_etcd_1:2379/health: dial tcp: lookup Kubernetes_etcd_1 on 169.254.169.250:53: no such host
7/19/2016 9:52:18 AMGet http://Kubernetes_etcd_1:2379/health: dial tcp: lookup Kubernetes_etcd_1 on 169.254.169.250:53: no such host
7/19/2016 9:52:21 AMGet http://Kubernetes_etcd_1:2379/health: dial tcp: lookup Kubernetes_etcd_1 on 169.254.169.250:53: no such host
7/19/2016 9:52:25 AMGet http://Kubernetes_etcd_1:2379/health: dial tcp: lookup Kubernetes_etcd_1 on 169.254.169.250:53: no such host
7/19/2016 9:52:29 AMGet http://Kubernetes_etcd_1:2379/health: dial tcp: lookup Kubernetes_etcd_1 on 169.254.169.250:53: no such host
7/19/2016 9:52:33 AMGet http://Kubernetes_etcd_1:2379/health: dial tcp: lookup Kubernetes_etcd_1 on 169.254.169.250:53: no such host
7/19/2016 9:52:38 AMGet http://Kubernetes_etcd_1:2379/health: dial tcp: lookup Kubernetes_etcd_1 on 169.254.169.250:53: no such host
7/19/2016 9:52:43 AMGet http://Kubernetes_etcd_1:2379/health: dial tcp: lookup Kubernetes_etcd_1 on 169.254.169.250:53: no such host
```
Cluster health on the healthy etcd host
=============================
bash-4.3# etcdctl cluster-health
member 5600de87e901 is healthy: got healthy result from http://10.42.136.198:237
9
cluster is healthy
bash-4.3#

**Comment by cloudnautique** [comment]
There is a race condition, that restarting the two nodes will fix. What happens is the first container comes up and sees itself, and the old metadata for the second container that is being restored. Then the second container comes up and sees that the first container that was recreated is of a lower index, but doesn't realize its never going to come up. Starting / Stopping both containers gets them a consistent up2date view of metadata. 

The reason this becomes a problem is that we continually probe with stale metadata and that needs to be reset, which is accomplished by the restart.

**Comment by LLParse** [comment]
Per offline discussion, we are going to solve this by adding a `--timeout` flag to giddyup probes so the container will eventually restart itself without user intervention

**Comment by soumyalj** [comment]
Same steps as above were performed on v1.1.2 using https://github.com/LLParse/testing-catalog
After disaster recovery, the two etcds did not come up. They were stuck in initializing state. We performed another DR,  etcds did not recover.

**Comment by LLParse** [comment]
Fixed in `rancher/etcd:v2.3.7-3`, please re-test.

**Comment by soumyalj** [comment]
Tested using https://github.com/LLParse/testing-catalog with etcd:v2.3.7 4. etcds came up successfully on the two new hosts after deleting the hosts in reconnecting state and performing disaster recovery.
